packages/app/provider/index.tsx

```diff
@@ -1,87 +1,129 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { syncSettings, updateTheme, useThemeSettings } from '@tloncorp/shared';
+import { subscribeToSettings } from '@tloncorp/shared/api';
 import { themeSettings } from '@tloncorp/shared/db';
-import { useEffect, useState } from 'react';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { TamaguiProvider, TamaguiProviderProps } from 'tamagui';
-import type { ThemeName } from 'tamagui';
 
 import { useIsDarkMode } from '../hooks/useIsDarkMode';
+import { AppTheme } from '../types/theme';
 import { config } from '../ui';
+import { getDisplayTheme, normalizeTheme } from '../ui/utils/themeUtils';
 
 export const ThemeContext = React.createContext<{
-  setActiveTheme: (theme: ThemeName) => void;
-  activeTheme: ThemeName;
+  setActiveTheme: (theme: AppTheme) => void;
+  activeTheme: AppTheme;
 }>({ setActiveTheme: () => {}, activeTheme: 'light' });
 
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false,
+      retry: false,
+    },
+  },
+});
+
 export function Provider({
   children,
   ...rest
 }: Omit<TamaguiProviderProps, 'config'>) {
+  return (
+    <QueryClientProvider client={queryClient}>
+      <ThemeProviderContent tamaguiProps={rest}>
+        {children}
+      </ThemeProviderContent>
+    </QueryClientProvider>
+  );
+}
+
+function ThemeProviderContent({
+  children,
+  tamaguiProps,
+}: {
+  children: React.ReactNode;
+  tamaguiProps: Omit<TamaguiProviderProps, 'config'>;
```

Comment:
Was following the pattern above where we omit 'config' from TamaguiProviderProps because we're providing our own config import and passing it in directly.